src/view/components/microbit/MicrobitSimulator.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from "react";
+import CONSTANTS, { DEVICE_LIST_KEY } from "../../constants";
+import PlayLogo from "../../svgs/play_svg";
+import StopLogo from "../../svgs/stop_svg";
+import Dropdown from "../Dropdown";
+import ActionBar from "../simulator/ActionBar";
+import { MicrobitImage } from "./MicrobitImage";
+
+const initialLedState = [
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+];
+
+interface vscode {
+    postMessage(message: any): void;
+}
+
+declare const vscode: vscode;
+
+const sendMessage = (type: string, state: any) => {
+    vscode.postMessage({ command: type, text: state });
+};
+interface IState {
+    active_editors: string[];

```

Comment:
Lets review everything that is a part of state in person, We should be very careful while deciding what to put in state as a change of same leads to re render and we don't want unnecessary renders

Reply:
Also sometimes it is good to define the shouldcomponentupdate in your component